src/protocol/tx-validity.md

```diff
@@ -112,48 +121,127 @@ def sum_predicate_gas_used(tx) -> int:
             total += input.predicateGasUsed
     return total
 
-"""
-Returns any minted amounts by the transaction
-"""
+    
+def sum_tx_bytes_gas_used(tx) -> int:
+    """
+    Computes the intrinsic gas cost of a transaction based on size in bytes
+    """
+    size(tx) * GAS_PER_BYTE
+
+
 def minted(tx, asset_id) -> int:
+    """
+    Returns any minted amounts by the transaction
+    """
     if tx.type != TransactionType.Mint or asset_id != tx.mint_asset_id:
         return 0
     return tx.mint_amount
 
+
 def sum_outputs(tx, asset_id) -> int:
     total: int = 0
     for output in tx.outputs:
         if output.type == OutputType.Coin and output.asset_id == asset_id:
             total += output.amount
     return total
 
+
+def sum_input_intrinsic_fees(tx) -> int:
+    """
+    Computes the intrinsic gas cost of verifying input utxos
+    """
+    total: int = 0
+    witness_indices = set(())
+    for input in tx.inputs:
+        # add fees required to read utxo from state
+        total += state_read_fee()
+        if input.type == InputType.Coin or input.type == InputType.Message:
+            # add fees required to remove coin or message from state
+            # note: this does not apply to contract inputs, as they are updated in place by the output
+            total += state_delete_fee()
+            # add fees allocated for predicate execution
+            total += gas_to_fee(input.predicateGasUsed, tx.gasPrice)
+            if input.predicateLength == 0:
+                # notate witness index if input is signed
+                witness_indices.add(input.witnessIndex)
+            else:
+                # add intrinsic cost of predicate merkleization based on number of predicate bytes
+                total += contract_root_bytes_fee(input.predicateLength)
+                # add intrinsic cost of vm initialization
+                total += vm_initialization_fee()
```

Comment:
The function `sum_predicate_gas_used` adds the `predicateGasUsed` to the total, and it is outside this function (`sum_input_intrinsic_fees`) since this function is concerned with intrinsic fees. 

However, I'm not sure that that kind of separation is really necessary, and the codebase sums `predicateGasUsed` and the contract root cost, so I am changing that here (and removing `intrinsic` from the name, since it now depends on the non-intrinsic `predicateGasUsed`).